### DISCUSSION TITLE
Can provide an example that includes Navigator?

### DISCUSSION THREAD

**Discussion by 18420621** [ethanblake4/flutter_eval#35] [General]
Can provide an example that includes Navigator?
I've been trying for a long time, but can't succeed.

**Comment by ethanblake4** [comment]
For now there is only support for named routes, eg `Navigator.of(context).pushNamed('/home')`... have you tried that?

**Reply by 18420621** [reply]
flutter 3.3.0
dart_eval: ^0.5.4
flutter_eval: ^0.5.3

----------------------------
 import 'package:flutter/material.dart';

              class MyApp extends StatelessWidget {
                const MyApp({super.key});

                // This widget is the root of your application.
                @override
                Widget build(BuildContext context) {
                  return MaterialApp(
                    title: 'flutter_eval demo',
                    home: const MyHomePage(title: 'flutter_eval demo home page'),
                     onGenerateRoute: (settings) {
        WidgetBuilder builder;
        switch(settings.name) {
          case '/':
            builder = (_) => MyHomePage(title: '',);
            break;
          case '/welcome':
            builder = (_) => welcome();
            break;
          default:
            throw new Exception('路由名稱有誤: \${settings.name}');
        }
        return new MaterialPageRoute(builder: builder, settings: settings);
      } ,
                  );
                }
              }

              class MyHomePage extends StatefulWidget {
                const MyHomePage({Key? key, required this.title}) : super(key: key);

                final String title;

                @override
                State<MyHomePage> createState() => _MyHomePageState();
              }

              class _MyHomePageState extends State<MyHomePage> {
                _MyHomePageState();
                int _counter = 0;

                @override
                Widget build(BuildContext context) {
                  return Scaffold(
                    appBar: AppBar(
                      title: Text(widget.title),
                    ),
                    body: Center(
                      child: Column(
                        mainAxisAlignment: MainAxisAlignment.center,
                        crossAxisAlignment: CrossAxisAlignment.center,
                        children: <Widget>[
                          const Text(
                            'You have pushed the button this many times:',
                          ),
                          Text(
                            '\$_counter',
                            style: Theme.of(context).textTheme.headline4,
                          ),
                        ],
                      )
                    ),
                    floatingActionButton: FloatingActionButton(
                       onPressed: (){
         Navigator.of(context).pushNamed('/welcome');
        },
                      tooltip: 'Increment',
                      child: Icon(Icons.add),
                    ),
                  );
                }
              }
              
class welcome extends StatefulWidget {

  @override
  State<welcome> createState() => _welcome();
}

class _welcome extends State<welcome> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Container(color: Colors.lightBlue,);
  }
}
----------------------------
This is the current state, sorry for not indenting

**Reply by 18420621** [reply]
run funtion Navigator.of(context).pushNamed('/welcome');

======== Exception caught by gesture ===============================================================
The following RuntimeException was thrown while handling a gesture:
dart_eval runtime exception: Could not find a generator for route RouteSettings("/welcome", null) in the _WidgetsAppState.
Make sure your root app widget has provided a way to generate
this route.
Generators for routes are searched for in the following order:
 1. For the "/" route, the "home" property, if non-null, is used.
 2. Otherwise, the "routes" table is used, if it has an entry for the route.
 3. Otherwise, onGenerateRoute is called. It should return a non-null value for any valid route not handled by "home" and "routes".
 4. Finally if all else fails onUnknownRoute is called.
Unfortunately, onUnknownRoute was not set.
#0      _WidgetsAppState._onUnknownRoute.<anonymous closure> (package:flutter/src/widgets/app.dart:1377:9)
#1      _WidgetsAppState._onUnknownRoute (package:flutter/src/widgets/app.dart:1392:6)
#2      NavigatorState._routeNamed (package:flutter/src/widgets/navigator.dart:4064:37)

RUNTIME STATE
=============
Program offset: 34342
Stack sample: [L0: [L0: $State$bridge<StatefulWidget>#b5b15, L1: Instance of '$BuildContext', L2: null], L1: 1, L2: Instance of '$BuildContext', L3: Instance of '$NavigatorState', L4: $"/welcome", L5: null, L6: null, L7: null, L8: null, L9: null]
Call stack: [0, -1, -1, -1]
TRACE:
34336: PushReturnValue ()
34337: PushConstant (C5)
34338: BoxString (L4)
34339: PushNull ()
34340: PushArg (L4)
34341: PushArg (L5)
34342: InvokeDynamic (L3.pushNamed)  <<< EXCEPTION
34343: PushReturnValue ()
34344: Pop (6)
34345: Pop (1)
![截圖 2023-01-05 下午12 26 26](https://user-images.githubusercontent.com/6592620/210701664-9872035e-9b62-42c2-86c2-57d3862e96f0.png)

**Reply by ethanblake4** [reply]
No support for `onGenerateRoute` yet, but there is support for `routes` so you should use that instead.

**Reply by 18420621** [reply]
Ready to use Thank you for your support
